core/server/models/post.js

```diff
@@ -137,13 +137,22 @@ Post = ghostBookshelf.Model.extend({
             // Variables to make the slug checking more readable
             newTitle    = this.get('title'),
             newStatus   = this.get('status'),
+            olderStatus = this.previous('status'),
             prevTitle   = this._previousAttributes.title,
             prevSlug    = this._previousAttributes.slug,
             tagsToCheck = this.get('tags'),
             publishedAt = this.get('published_at'),
             tags = [];
 
-        // both page and post can get scheduled
+        // CASE: disallow published -> scheduled
+        // @TODO: remove when we have versioning based on updated_at
+        if (newStatus !== olderStatus && newStatus === 'scheduled' && olderStatus === 'published') {
+            return Promise.reject(new errors.ValidationError(
+                i18n.t('errors.models.post.isAlreadyPublished', {key: 'status'})
+            ));
+        }
```

Comment:
I think there is a missing guard that should either be here or in the scheduling api - there's currently nothing that's checking that the scheduler is doing the right thing? (Or I haven't found it yet).

What I mean is, I would expect that if the scheduler endpoint receives a request to /post/:id/ that something would check that the post is still currently set to "scheduled" and the publish_at time is either in the past or somewhere in the near future, so that the post isn't getting scheduled incorrectly.

The case I'm thinking of, is someone unscheduling, or rescheduling a post just a few seconds before the schedule time, and the scheduler not getting the event & cancelling its job in time.